templates/keycloak/files/.platform.app.yaml

```diff
@@ -0,0 +1,83 @@
+        JDBC_MARIADB_VERSION: 2.2.3
+        LAUNCH_JBOSS_IN_BACKGROUND: 1
+        #DB_VENDOR: mariadb
+hooks:
+    build: |
+        set -e
+
+        if [ ! -d "$PLATFORM_CACHE_DIR"/keycloak-$KC_VERSION ]; then
+            echo "Downloading KeyCloak version ${KC_VERSION}..."
+            (
+                cd "$PLATFORM_CACHE_DIR"
+                curl -L -o $KC_VERSION.tar.gz https://downloads.jboss.org/keycloak/$KC_VERSION/keycloak-$KC_VERSION.tar.gz
+                tar -zxf $KC_VERSION.tar.gz
+                rm -f $KC_VERSION.tar.gz
+            )
+        else
+            echo "Using cached archive of Keycloak: ${KC_VERSION}..."
+        fi
+
+        echo "Installing archive of Keycloak: ${KC_VERSION}..."
+        cp -R "$PLATFORM_CACHE_DIR"/keycloak-"$KC_VERSION"/* .
+        mv standalone/configuration standalone/cfg.bkp
+
+        curl -L https://repo1.maven.org/maven2/org/mariadb/jdbc/mariadb-java-client/$JDBC_MARIADB_VERSION/mariadb-java-client-$JDBC_MARIADB_VERSION.jar > modules/system/layers/base/org/mariadb/jdbc/main/mariadb-jdbc.jar
+        
+        rm -rf standalone/tmp/auth
+        rm -rf domain/tmp/auth
+    
+    deploy: |
+        if [ -f tmp/INSTALLED ]; then
+            cp standalone/cfg.bkp/* standalone/configuration/
+            cp -f tools/standalone.xml standalone/configuration/standalone.xml
+            cp -f tools/logging.properties standalone/configuration/logging.properties
+            bin/add-user-keycloak.sh -r master --user "admin" --password "password"
+            touch tmp/INSTALLED
+        fi;
+        export DB_PORT=`echo $PLATFORM_RELATIONSHIPS|base64 -d|json_pp|./jq -r ".database[0].port"`
+        export DB_ADDR=`echo $PLATFORM_RELATIONSHIPS|base64 -d|json_pp|./jq -r ".database[0].host"`
+        export DB_PASSWORD=`echo $PLATFORM_RELATIONSHIPS|base64 -d|json_pp|./jq -r ".database[0].password"`
+        export DB_USER=`echo $PLATFORM_RELATIONSHIPS|base64 -d|json_pp|./jq -r ".database[0].username"`
+        export DB_DATABASE=`echo $PLATFORM_RELATIONSHIPS|base64 -d|json_pp|./jq -r ".database[0].path"`
+        export FRONTEND_URL=`echo $PLATFORM_ROUTES|base64 -d|json_pp|./jq -r 'to_entries | map_values(.value + { slug: .key }) | .[] | select(.primary == true) | .slug'`
+
+relationships:
+    database: 'database:mariadb'
+
+web:
+    upstream:
+        socket_family: tcp
+        protocol: http
+    commands:
+        start: |
+            bin/standalone.sh -Dkeycloak.frontendUrl=$FRONTEND_URL -Djboss.http.port=$PORT
+    locations:
+        /:
+            passthru: true
+
+mounts:
+    'standalone/data':
+        source: local
+        source_path: data
+    'standalone/configuration':
+        source: local
+        source_path: configuration
+    'standalone/log':
+        source: local
+        source_path: log
+    'standalone/deployments':
+        source: local
+        source_path: deployments
+    'standalone/tmp':
+        source: local
+        source_path: tmp
+
+disk: 1024
```

Comment:
Total nitpick: I prefer to put the `disk` key before `mounts`.  I think that's more common in the other templates.